modules/rostests/apitests/shell32/ExtractIconEx.cpp

```diff
@@ -22,16 +24,72 @@ EXTRACTICONTESTS IconTests[] =
     {L"%SystemRoot%\\System32\\autochk.exe", 0},
 
     /* Non-existing files */
-    {L"%SystemRoot%\\non-existent-file.sdf", 0}
+    {L"%SystemRoot%\\non-existent-file.sdf", 0},
+
+    /* Multiple icons in the same EXE file (18 icons) */
+    {L"%SystemRoot%\\explorer.exe", 18},
+
+    /* Multiple icons in the same ICO file (6 icons)
+     * Per MS: If the file is an .ico file the return value is 1." */
+    {L"sysicon.ico", 1},
+
+    /* ICO file with both normal and PNG icons */
+    {L"ROS.ico", 1}
 };
 
 START_TEST(ExtractIconEx)
 {
-    UINT i, nReturnedIcons;
+    UINT i, nReturnedIcons, nExtractedIcons, iSize;
+    HRSRC hRes;
+    HGLOBAL hData;
+    LPVOID lpResLock;
+    LPCSTR sData;
+    FILE *fout;
+    CHAR FileName[2][13] = { "ROS.ico", "sysicon.ico" };
+
+    for (i = 2; i < 4; i++) // Point to resources 2 and 3
```

Comment:
Could you split this task (extracting embeded data) into newly defined functions?

Reply:
@katahiromz, I have split out the extraction of embedded data and added deleting of the files as you asked. Thanks.